specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,658 @@
+  /// the stop request event so that the callback of the next start request
+  /// should register the textures again.
+  /// The event is triggered when all requests for given stream id closed
+  /// by the Javascript, or the host's Stop API call.
+  /// texture related API calls after this event will return an error
+  /// of HRESULT_FROM_WIN32(ERROR_INVALID_STATE).
+  HRESULT add_Stopped(
+      [in] ICoreWebView2StagingTextureStreamStoppedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for Stopped event.
+  HRESULT remove_Stopped(
+      [in] EventRegistrationToken token);
+  /// Creates texture that will be referenced by the host and the browser.
+  /// By using the texture mechanism, the host does not have to
+  /// send respective texture to the renderer, instead it notifies it
+  /// with internal texture id, which is the identity of the texture.
+  /// The texture is 2D texture, IDXGIResource, format and will be
+  /// exposed through shared HANDLE or IUnknown type through ICoreWebView2StagingTexture.
+  /// Whenever the host has new texture to write, it should ask
+  /// reusable ICoreWebView2StagingTexture from the GetAvailableTexture,
+  /// which returns ICoreWebView2StagingTexture.
+  /// If the GetAvailableTexture returns an error, then the host calls the
+  /// CreateTexture to allocate new texture.
+  /// The API also registers created shared handle to the browser once it
+  /// created the resource.
+
+  /// Unit for width and height is texture unit (in texels).
+  /// 'https://learn.microsoft.com/en-us/windows/win32/api/d3d11ns-d3d11-d3d11_texture2d_desc'
+  HRESULT CreateTexture(
+    [in] UINT32 widthInTexels,
+    [in] UINT32 heightInTexels,
+    [out, retval] ICoreWebView2StagingTexture** texture);
+  /// Returns reuseable texture for video frame rendering.
+  /// Once the renderer finishes rendering of texture's video frame, which
+  /// was requested by Present, the renderer informs the host so that it can
+  /// be reused. The host has to create new texture with CreateTexture
+  /// if the API return an error HRESULT_FROM_WIN32(ERROR_NO_MORE_ITEMS).
+  HRESULT GetAvailableTexture([out, retval] ICoreWebView2StagingTexture** texture);
+  /// Removes texture when the host removes the backed 2D texture.
+  /// The host can save the resources by deleting 2D textures
+  /// when it changes the frame sizes. The API will send a message
+  /// to the browser where it will remove texture.
+  HRESULT CloseTexture([in] ICoreWebView2StagingTexture* texture);
+  /// Sets rendering image/resource through ICoreWebView2StagingTexture.
+  /// The texture must be retrieved from the GetAvailableTexture or
+  /// created via CreateTexture.
+  /// It is expected that hhe host writes new image/resource to the local
+  /// shared 2D texture of the texture (handle/resource).
+  HRESULT PresentTexture([in] ICoreWebView2StagingTexture* texture)
+
+  /// Stop streaming of the current stream id.

```

Comment:
COREWEBVIEW2_TEXTURE_STREAM_ERROR_NO_VIDEO_TRACK_STARTED is from the enum not an HRESULT right? We should document the HRESULT that gets returned if you call one of the methods while the texture stream is stopped.

```suggestion
  /// Stops this texture stream from streaming and moves it into the stopped state. When moving to the stopped state the `ICoreWebView2TextureStream Stopped` event will be raised and in script the `MediaStreamTrack ended` event will be raised. Once stopped, script may again call `Webview.getTextureStream` moving the texture stream back to the start requested state. See the `StartRequested` event for details.
  /// Once stopped, calls to CreateTexture, GetAvailableTexture, PresentTexture, and CloseTexture will fail with COREWEBVIEW2_TEXTURE_STREAM_ERROR_NO_VIDEO_TRACK_STARTED. See those methods for details.
  /// The `Stop` method is implicitly called when the texture stream object is destroyed.
```